modules/packages/freebsd_ports

```diff
@@ -0,0 +1,74 @@
+#!/bin/sh -e
+
+command=$1
+
+while read -r line; do
+  export INPUT_$line
+done
+
+list_updates() {
+  pkg version -v -l "<" | sed -e 's/\([^ ]*\)-[^-]* .* \(.*\))/Name=\1\
+Version=\2\
+Architecture=none/'
+}
+
+case $command in
+  supports-api-version)
+    echo 1
+  ;;
+  get-package-data)
+    name="${INPUT_File?File must be given to get-package-data}"
+    echo PackageType=repo
+    echo Name=$name
+  ;;
+  list-installed)
+    pkg info -f -a | egrep '^(Name|Version|Architecture)' | sed -e 's/[ ]*:[ ]*/=/' -e 's/^Architecture=.*/Architecture=none/'
```

Comment:
Concur - separate function would be better when considering possibility of pkg lock.